src/server.rs

```diff
@@ -117,6 +118,60 @@ impl LanguageServer for KotoServer {
         }
     }
 
+    async fn hover(&self, params: HoverParams) -> Result<Option<Hover>> {
+        let uri = params.text_document_position_params.text_document.uri;
+        let position = params.text_document_position_params.position;
+
+        let lock_await = self.source_info.lock().await;
+        let result = lock_await.get(&uri).and_then(|info| {
+            info.get_referenced_definition_location(position)
+                .and_then(|location| {
+                    if location.uri.as_ref() == &uri {
+                        info.get_definition_from_location(location)
+                            .map(|definition| {
+                                let symbol = DocumentSymbol::from(&definition);
+                                format!("**{}**  \n{:?} reference", symbol.name, symbol.kind,)
```

Comment:
You can pull `name` and `kind` directly from `definition` without converting to `DocumentSymbol`.

Reply:
Done